src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1128,8 +1128,25 @@
             (munge-uuid "transaction_uuid") ", environment_id, status_id
             FROM reports"))
 
+    (sql/do-commands
```

Comment:
actually, now that I consider it you may be right. it's not certnames that's the issue, it's resource_events, which is a massive table in the cases that this patch addresses.
As written this creates a new column on that table at the same time that we are already recreating it, but the really expensive part is creating a new index, and we gain nothing from having that mixed in with this other stuff.